### ISSUE TITLE
Service Binding Operator Upgrade from v0.8.0 to v0.9.0 fails due to missing spec.application

### ISSUE THREAD

**Issue by jcperera-github** [redhat-developer/service-binding-operator#1000]
### What is the environment (Minikube, Openshift)?
OpenShift

### What is the SBO version used?
0.9.0

### What are the steps to reproduce this issue?
  1. Automatic Upgrade from SBO v0.8.0 to v0.9.0

### What is the expected behaviour?
No errors

### What is the actual behaviour?
Message: Failed to apply object: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"ServiceBinding.binding.operators.coreos.com \\"demo01-suite-binding\\" is invalid: spec.application: Required value","reason":"Invalid","details":{"name":"demo01-suite-binding","group":"binding.operators.coreos.com","kind":"ServiceBinding","causes":[{"reason":"FieldValueRequired","message":"Required value","field":"spec.application"}]},"code":422}\n'

### Additional Information (Screenshots, etc)
Contents of the actual spec node that worked well on v.0.8.0:

![image](https://user-images.githubusercontent.com/59753450/127167654-d0bfe972-06ce-4daa-9a00-a2781c11e354.png)

**Comment by pmacik** [comment]
@jcperera-github This is because in `v0.9.0` the binding with no application was dropped by https://github.com/redhat-developer/service-binding-operator/pull/977 - see https://github.com/redhat-developer/service-binding-operator/issues/897

**Comment by jcperera-github** [comment]
Ok, it broke backwards compatibility... I cannot seem to downgrade to v0.8.0 due to that version not existing on the Catalog Source for OpenShift any longer... 

Any suggestions?

**Comment by jcperera-github** [comment]
Ok, I have removed v0.9.0 from the Installed Operators (from the Catalog) and manually installed the v0.8.0 using the Command Line:
`oc apply -f https://github.com/redhat-developer/service-binding-operator/releases/download/v0.8.0/release.yaml`
And that fixed the issue for us. The code for the bindings is not ours, but installed by an IBM product so we preferred not to edit the Service Binding definitions and manually downgrade instead.
Closing the ticket.
